mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/KeyValueBasedPersistenceProvider.java

```diff
@@ -394,18 +399,18 @@ public List<IMantisJobMetadata> loadAllJobs() throws IOException {
 
                 workersFoundCounter.increment();
                 for (MantisWorkerMetadataWritable workerMeta : workersByJobId.get(jobId)) {
-                    Preconditions.checkState(
-                        jobMeta.addWorkerMedata(workerMeta.getStageNum(), workerMeta, null),
-                        "JobID=%s stage=%d workerIdx=%d has existing worker, existing=%s, new=%s",
-                        workerMeta.getJobId(),
-                        workerMeta.getStageNum(),
-                        workerMeta.getWorkerIndex(),
-                        jobMeta.getWorkerByIndex(workerMeta.getStageNum(), workerMeta.getWorkerIndex()).getWorkerId(),
-                        workerMeta.getWorkerId());
+                    // If there are duplicate workers on the same stage index, only attach the one with latest
+                    // worker number. The stale workers will not present in the stage metadata thus gets terminated
+                    // when it sends heartbeats to its JobActor.
```

Comment:
It looks like this logic is being decided at the MantisJobWorkerMetadataWritable level. Could we instead do this here? We already have two different building blocks today - one to add and another to replace. Let's use them instead. 

It's also not clear to me what the boolean represents in this new method call.

Reply:
"one to add and another to replace,": i think it's just different names used for different components. I only find one path to add/replace workers in the code.

Reply:
I can do "addWorkerMedata" -> "tryAddOrReplaceWorker"